### PR TITLE
BeanValidationDiagnosticsCollector: Refactoring method and field annotation processing loops into one method.

### PR DESCRIPTION
**Pull request by mrglavas**
Resolves https://github.com/OpenLiberty/liberty-tools-intellij/issues/1070

### CODE REVIEW

**Review by turkeylurkey — APPROVED** [review]

File: src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/beanvalidation/BeanValidationDiagnosticsCollector.java [hunk 1]
```diff
@@ -35,40 +35,34 @@ protected String getDiagnosticSource() {
...
-                            validAnnotation(method, annotation, matchedAnnotation, diagnostics);
-                        }
-                    }
+                    processAnnotations(method, type, diagnostics);
                 }
             }
         }
+    }
 
+    private void processAnnotations(PsiJvmModifiersOwner psiModifierOwner, PsiClass type, List<Diagnostic> diagnostics) {
```
Comment (vaisakhkannan) [inline src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/beanvalidation/BeanValidationDiagnosticsCollector.java:57]: @mrglavas , Just to confirm, the `PsiJvmModifiersOwner` class can be used for both `PsiField` and `PsiMethod`, right?

Comment (vaisakhkannan) [inline src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/beanvalidation/BeanValidationDiagnosticsCollector.java:57]: Just to confirm, the PsiJvmModifiersOwner class can be used for both PsiField and PsiMethod, right?

Comment (mrglavas) [reply src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/beanvalidation/BeanValidationDiagnosticsCollector.java:57]: @vaisakhkannan Yes, that's a common interface that both of them implement.

Comment (vaisakhkannan) [reply src/main/java/io/openliberty/tools/intellij/lsp4jakarta/lsp4ij/beanvalidation/BeanValidationDiagnosticsCollector.java:57]: Ok @mrglavas , Thanks , I approved

**Review by vaisakhkannan — APPROVED** [review]

**Pull request merged**